### PR TITLE
cubeviz and mosviz documentation outline added

### PR DESCRIPTION
**Pull request by orifox**

### CODE REVIEW

File: docs/cubeviz/index.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+
+.. image:: ../logos/cubeviz.svg
+   :width: 400
+
+#######
+CubeViz
+#######
+
+CubeViz is an visualization and analysis toolbox for data cubes from integral field units (IFUs). It is built as part of the glue visualiztion tool. CubeViz is designed to work with data cubes from the NIRSpec and MIRI instruments on JWST, and will work with data cubes from any IFU. It uses the `specutils <https://specutils.readthedocs.io/en/latest/>`_ package from `Astropy <https://www.astropy.org>`_ .
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:9]: "a visualization"
"glue visualization" (sp.)
"will work with data cubes from any IFU"   Will it?

File: docs/cubeviz/index.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+The core functionality of CubeViz currently includes the ability to:
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:12]: "Cubeviz"

File: docs/cubeviz/index.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+* view the wavelength slices (RA, DEC) in a data cube,
+
+* view flux, error, and data quality slices simultaneously,
+
+* view spectra from selected spatial (RA, DEC) regions,
+
+* smooth cubes spactially (RA, DEC) and spectrally (wavelength), and
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:20]: "spatially" (sp.)

File: docs/cubeviz/index.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+* create and display contour maps.
+
+* collapse cubes over selected wavelength regions,
+
+* fit spectral lines,
+
+* create moment maps,
+
+* perform continuum subtraction,
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:30]: The ability to perform continuum subtraction is not yet implemented.

File: docs/cubeviz/index.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+* cube arithmetic,
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:32]: cube arithmetic is not yet available

File: docs/cubeviz/index.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+* overlay spectral line lists,
+
+* save edited cubes,
+
+* save figures,
+
+* mock slit observations,
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:40]: mock slit observations are not available

File: docs/cubeviz/index.rst [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+* accurate spectro-photometry,
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:42]: accurate spectrophotometry is not available

File: docs/cubeviz/index.rst [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+* fit models to every spaxel
+
+Future functionality will include the ability to:
+
+* save and restore a session,
+
+* create kinematic maps (rotation velocity and velocity dispersion),
+
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:50]: Moment maps are in this sprint

File: docs/cubeviz/index.rst [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+.. image:: ../logos/cubeviz.svg
+   :width: 400
+
+#######
+CubeViz
+#######
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:7]: 'Cubeviz'

File: docs/cubeviz/plugins.rst [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+===============
+
+More words...
+
+Model Fitting 
+=============
+
+More words...
+
+Countours
```
Comment (PatrickOgle) [inline docs/cubeviz/plugins.rst:22]: "Contours"

File: docs/index.rst [hunk 11]
```diff
@@ -18,21 +22,23 @@ contextual information like on-sky views of the spectrograph slit.
 CubeViz provides of view of spectroscopic data cubes (like those to be
 produced by JWST MIRI), along with 1D spectra extracted from the cube.
 
+Using Jdaviz
```
Comment (PatrickOgle) [inline docs/index.rst:25]: "Jdaviz"
"Cubeviz"

File: docs/mosviz/index.rst [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+######
+MosViz
+######
+
+MOSViz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with pipeline output: spectra and associated images, or just with spectra. MOSViz is created to work with data from any telescope/instrument, but is built with the micro-shutter array (MSA) on the JWST/NIRSpec spectrograph and the JWST/NIRCam imager in mind. As such, MOSViz has some features specific to NIRSpec and NIRCam data.
+
+The NIRSpec micro-shutter array (MSA) will produce ~100 spectra per pointing. Many users will perform surveys with the MSA that will result in data sets of thousands to tens of thousands of spectra. Especially in the early days of JWST, careful inspection of all of the data will be critical to achieving scientific goals, and it is important to make this task efficient. Inspection involves looking at the locations of astronomical sources within shutters, the location of background apertures in the observed field, the quality of the 2D spectra, and the quality of the 1D extracted spectra. It also often involves simple interactive measurements of quantities such as wavelengths, velocities, line fluxes, widths.
```
Comment (PatrickOgle) [inline docs/mosviz/index.rst:11]: "tens to hundreds of spectra"

I don't think DQA is really the purview of MOSviz.  We have the JWQL tools to handle that in-house. 
This should be focused on science-driven data inspection and analysis instead.

File: docs/specviz/index.rst [hunk 13]
```diff
@@ -1,9 +1,26 @@
...
+
+SpecViz will soon include the ability to:
+
+* Measure the average of multiple spectra, detrending, and apply Fourier filters.
+
+* Interactively renormalize data from spectral templates.
+
+* And more…
+
+**Using SpecViz**
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:23]: "Specviz"  several instances

File: docs/cubeviz/index.rst (hunk 3 above)
Comment (orifox) [reply docs/cubeviz/index.rst:20]: done

File: docs/cubeviz/index.rst (hunk 8 above)
Comment (PatrickOgle) [reply docs/cubeviz/index.rst:42]: kinematic maps are a subset of moment maps.

File: docs/cubeviz/index.rst [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+* save figures,
+
+* fit models to every spaxel
+
+Future functionality will include the ability to:
+
+* save and restore a session,
+
+* create kinematic maps (rotation velocity and velocity dispersion),
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:41]:
```suggestion

```

File: docs/cubeviz/index.rst [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+* smooth cubes spatially (RA, DEC) and spectrally (wavelength), and
+
+* create and display contour maps.
+
+* collapse cubes over selected wavelength regions,
+
+* fit spectral lines,
+
+* create moment maps,
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:27]:
```suggestion
* create moment maps, including line flux and kinematic maps (rotation velocity and velocity dispersion)
```

File: docs/mosviz/index.rst [hunk 16]
```diff
@@ -0,0 +1,22 @@
+
+.. image:: ../logos/mosviz.svg
+   :width: 400
+
+######
+Mosviz
+######
+
+MOSViz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with pipeline output: spectra and associated images, or just with spectra. MOSViz is created to work with data from any telescope/instrument, but is built with the micro-shutter array (MSA) on the JWST/NIRSpec spectrograph and the JWST/NIRCam imager in mind. As such, MOSViz has some features specific to NIRSpec and NIRCam data.
```
Comment (PatrickOgle) [inline docs/mosviz/index.rst:9]:
```suggestion
MOSviz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with pipeline output: spectra and associated images, or just with spectra. MOSviz is created to work with data from any telescope/instrument, but is built with the micro-shutter array (MSA) on the JWST/NIRSpec spectrograph and the JWST/NIRCam imager in mind. As such, MOSViz has some features specific to NIRSpec and NIRCam data.
```

File: docs/specviz/index.rst [hunk 17]
```diff
@@ -1,9 +1,26 @@
...
+
+.. image:: ../logos/specviz.svg
+   :width: 400
+
+#######
 Specviz
-*******
+#######
+
+SpecViz is a tool for visualization and quick-look analysis of 1D astronomical spectra. It is written in the Python programming language, and therefore can be run anywhere Python is supported (see Installation). SpecViz is built on top of the `specutils <https://specutils.readthedocs.io/en/latest/>`_ package from `Astropy <https://www.astropy.org>`_ , providing a visual, interactive interface to the analysis capabilities in that library.
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:9]:
```suggestion
Specviz is a tool for visualization and quick-look analysis of 1D astronomical spectra. It is written in the Python programming language, and therefore can be run anywhere Python is supported (see Installation). Specviz is built on top of the `specutils <https://specutils.readthedocs.io/en/latest/>`_ package from `Astropy <https://www.astropy.org>`_ , providing a visual, interactive interface to the analysis capabilities in that library.
```

File: docs/specviz/index.rst [hunk 18]
```diff
@@ -1,9 +1,26 @@
...
+
+SpecViz allows spectra to be easily plotted and examined. It supports flexible spectral unit conversions, custom plotting attributes, interactive selections, multiple plots, and other features.
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:11]:
```suggestion
Specviz allows spectra to be easily plotted and examined. It supports flexible spectral unit conversions, custom plotting attributes, interactive selections, multiple plots, and other features.
```

File: docs/specviz/index.rst [hunk 19]
```diff
@@ -1,9 +1,26 @@
...
+
+SpecViz notably includes a measurement tool for spectral lines which enables the user, with a few mouse actions, to perform and record measurements. It has a model fitting capability that enables the user to create simple (e.g., single Gaussian) or multi-component models (e.g., multiple Gaussians for emission and absorption lines in addition to regions of flat continua). A typical data-analysis workflow might involve data exploration using SpecViz and then scripting to create more complex measurements or modeling workflows using specutils.
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:13]:
```suggestion
Specviz notably includes a measurement tool for spectral lines which enables the user, with a few mouse actions, to perform and record measurements. It has a model fitting capability that enables the user to create simple (e.g., single Gaussian) or multi-component models (e.g., multiple Gaussians for emission and absorption lines in addition to regions of flat continua). A typical data-analysis workflow might involve data exploration using SpecViz and then scripting to create more complex measurements or modeling workflows using specutils.
```

File: docs/specviz/index.rst [hunk 20]
```diff
@@ -1,9 +1,26 @@
...
+
+SpecViz will soon include the ability to:
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:15]:
```suggestion
Specviz will soon include the ability to:
```

File: docs/specviz/index.rst [hunk 21]
```diff
@@ -1,9 +1,26 @@
...
+
+* Measure the average of multiple spectra, detrending, and apply Fourier filters.
+
+* Interactively renormalize data from spectral templates.
 
-Words go here...
+* And more…
 
+**Using SpecViz**
```
Comment (PatrickOgle) [inline docs/specviz/index.rst:23]:
```suggestion
**Using Specviz**
```

File: docs/cubeviz/notebook.rst [hunk 22]
```diff
@@ -0,0 +1,6 @@
+***********************************
+Using CubeViz in a Jupyter Notebook
```
Comment (PatrickOgle) [inline docs/cubeviz/notebook.rst:2]:
```suggestion
Using Cubeviz in a Jupyter Notebook 
```

File: docs/cubeviz/index.rst [hunk 23]
```diff
@@ -0,0 +1,61 @@
+
+.. image:: ../logos/cubeviz.svg
+   :width: 400
+
+#######
+Cubeviz
+#######
+
+Cubeviz is a visualization and analysis toolbox for data cubes from integral field units (IFUs). It is built as part of the `Glue visualiztion <https://glueviz.org>`_ tool. CubeViz is designed to work with data cubes from the NIRSpec and MIRI instruments on JWST, and will work with IFU data cubes. It uses the `specutils <https://specutils.readthedocs.io/en/latest/>`_ package from `Astropy <https://www.astropy.org>`_ .
```
Comment (PatrickOgle) [inline docs/cubeviz/index.rst:9]:
```suggestion
Cubeviz is a visualization and analysis toolbox for data cubes from integral field units (IFUs). It is built as part of the `Glue visualiztion <https://glueviz.org>`_ tool. Cubeviz is designed to work with data cubes from the NIRSpec and MIRI instruments on JWST, and will work with IFU data cubes. It uses the `specutils <https://specutils.readthedocs.io/en/latest/>`_ package from `Astropy <https://www.astropy.org>`_ .
```

File: docs/mosviz/index.rst [hunk 24]
```diff
@@ -0,0 +1,22 @@
+
+.. image:: ../logos/mosviz.svg
+   :width: 400
+
+######
+Mosviz
+######
+
+MOSviz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with pipeline output: spectra and associated images, or just with spectra. MOSviz is created to work with data from any telescope/instrument, but is built with the micro-shutter array (MSA) on the JWST/NIRSpec spectrograph and the JWST/NIRCam imager in mind. As such, MOSViz has some features specific to NIRSpec and NIRCam data.
```
Comment (PatrickOgle) [inline docs/mosviz/index.rst:9]:
```suggestion
MOSviz is a quick-look analysis and visualization tool for multi-object spectroscopy (MOS). It is designed to work with pipeline output: spectra and associated images, or just with spectra. MOSviz is created to work with data from any telescope/instrument, but is built with the micro-shutter array (MSA) on the JWST/NIRSpec spectrograph and the JWST/NIRCam imager in mind. As such, MOSviz has some features specific to NIRSpec and NIRCam data.
```

**Comment by nmearl** [comment]
The repeated commits should be squashed, and the pr rebased to remove the merge commits.

**Pull request merged**